File: CodeForces/1-100/4/C/4C.rb
# Problem 4C: Registration System
# https://codeforces.com/contest/4/problem/C
# Executed with: ruby 3.3.8 (2025-04-09 revision b200bad6cd) [aarch64-linux-gnu]
# Linux version: Linux kernel 6.17.0-5-generic
# Submitted on: October 25th, 2025
# Codeforces language used: Ruby 3.2.2

history = {}
usernames = gets.to_i

for _ in 1..usernames

    username = gets.chomp

    if history.key?(username)
        puts username + history[username].to_s
        history[username] += 1
    
    else
        puts "OK"
        history[username] = 1
    end
end

File: CodeForces/1-100/4/C/4C.py
# Problem 4C: Registration System
# https://codeforces.com/contest/4/problem/C
# Executed with: Python 3.11.2
# Linux version: Linux kernel 6.1.0-10-amd64
# Submitted on: September 27th, 2024
# Codeforces language used: PyPy 3.10 (7.3.15, 64bit)

history = {}
usernames = int(input())
for _ in range(0,usernames):
    username = str(input())
    if username in history:
        print(f"{username}{history[username]}")
        history[username] += 1
    else:
        print("OK")
        history[username] = 1
